nixos/modules/services/security/vaultwarden/default.nix

```diff
@@ -213,6 +248,27 @@ in {
         User = lib.mkDefault user;
         Group = lib.mkDefault group;
         ExecStart = "${pkgs.bash}/bin/bash ${./backup.sh}";
+        ReadOnlyDirectories = lib.mkIf (!lib.hasPrefix configEnv.DATA_FOLDER cfg.backupDir) [ configEnv.DATA_FOLDER ];
+        ReadWriteDirectories = [ "-${cfg.backupDir}" ];
+
+        # Hardening
+        LockPersonality = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
```

Comment:
PrivateMounts is not in https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/security/vaultwarden/default.nix#L183-L219 which I think is alright